server/templates/ga.html

```diff
@@ -1,8 +1,25 @@
+<script>
+  // detecting "do not track" via these instructions:
+  // https://dev.to/corbindavenport/how-to-correctly-check-for-do-not-track-with-javascript-135d
+  if (window.doNotTrack == "1" ||
+      navigator.doNotTrack == "yes" || navigator.doNotTrack == "1" ||
+      navigator.msDoNotTrack == "1" ||
+      ('msTrackingProtectionEnabled' in window.external &&
+       window.external.msTrackingProtectionEnabled())) {
+    // do not track is enabled: disable per google's instructions:
+    // https://developers.google.com/analytics/devguides/collection/gtagjs/user-opt-out
+    window['ga-disable-{{ GA_TRACKING_ID }}'] = true;
+  }
+</script>
 <script async src="https://www.googletagmanager.com/gtag/js?id={{ GA_TRACKING_ID }}"></script>
 <script>
-  window.dataLayer = window.dataLayer || [];
-  function gtag(){dataLayer.push(arguments);}
-  gtag('js', new Date());
+  if (!window['ga-disable-{{ GA_TRACKING_ID }}']) {
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
+    gtag('js', new Date());
 
-  gtag('config', '{{ GA_TRACKING_ID }}');
+    gtag('config', '{{ GA_TRACKING_ID }}');
+  } else {
+    console.log('Do not track enabled, google analytics disabled')
```

Comment:
Trying to do some basic local testing: I don't get this console message with "Do Not Track" turned on in my browser and hitting the local server.  But maybe GA never comes into play with the local server anyway?

Reply:
Right, we don't include any google analytics code unless the GA_TRACKING_ID environment variable is set:

https://github.com/iodide-project/iodide/blob/1f9ac4b0e25299f76727e49e05c93c28de191482/server/context_processors.py#L9
https://github.com/iodide-project/iodide/blob/1f9ac4b0e25299f76727e49e05c93c28de191482/server/settings.py#L55

Reply:
Ok, in that case, :ship: it, and we'll come back around and test it on Heroku once it's deployed.